modules/aws/vpc.go

```diff
@@ -143,6 +143,43 @@ func GetSubnetsForVpcE(t *testing.T, vpcID string, region string) ([]Subnet, err
 	return subnets, nil
 }
 
+// IsPublicSubnet returns True if the subnet identified by the given id in the provided region is public.
+func IsPublicSubnet(t *testing.T, subnetId string, region string) bool {
+	isPublic, err := IsPublicSubnetE(t, subnetId, region)
+	require.NoError(t, err)
+	return isPublic
+}
+
+// IsPublicSubnetE returns True if the subnet identified by the given id in the provided region is public.
+func IsPublicSubnetE(t *testing.T, subnetId string, region string) (bool, error) {
+	subnetIdFilterName := "association.subnet-id"
+
+	subnetIdFilter := ec2.Filter{
+		Name:   &subnetIdFilterName,
+		Values: []*string{&subnetId},
+	}
+
+	client, err := NewEc2ClientE(t, region)
+	if err != nil {
+		return false, err
```

Comment:
Not sure if this is the best way of handling optional booleans...

Reply:
This is fine. In idiomatic go, you would check the `err` first and disregard the boolean if there is an error.